### PR TITLE
RsXSLT.consume() should close InputStream

### PR DESCRIPTION
**Pull request by Happy-Neko**
Test coverage for #441.

@davvd Please proceed with code review.

### CODE REVIEW

**Comment by davvd** [comment]
@happy-neko Many thanks, I will find someone to review it before we merge

**Comment by davvd** [comment]
@carlosmiranda it's yours, please review

File: src/test/java/org/takes/rs/RsXSLTTest.java [hunk 1]
```diff
@@ -125,6 +126,46 @@ public Source resolve(final String href,
     }
 
     /**
+     * RsXSLT closes decorated Response body's InputStream when XML conversion
+     * is done.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void closesDecoratedResponseInputStream() throws IOException {
```
Comment (carlosmiranda) [inline src/test/java/org/takes/rs/RsXSLTTest.java]: @Happy-Neko let's declare throws `Exception` here. See [this article](http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names).

File: src/test/java/org/takes/rs/RsXSLTTest.java [hunk 2]
```diff
@@ -125,6 +126,46 @@ public Source resolve(final String href,
...
+        final StateAwareInputStream inputStream =
+            new StateAwareInputStream(IOUtils.toInputStream(xml));
+        MatcherAssert.assertThat(
+            new RsPrint(
+                new RsXSLT(
+                    new RsText(inputStream),
+                    new URIResolver() {
+                        @Override
+                        public Source resolve(final String href,
+                                              final String base) {
```
Comment (carlosmiranda) [inline src/test/java/org/takes/rs/RsXSLTTest.java]: @Happy-Neko Too much indentation, should be 4 lines indented from line 153 only.

**Comment by carlosmiranda** [comment]
@Happy-Neko A couple of comments above.

**Comment by Happy-Neko** [comment]
@carlosmiranda Thanks for the review. Fix committed.

**Comment by carlosmiranda** [comment]
@rultor merge please

File: src/test/java/org/takes/rs/RsXSLTTest.java [hunk 3]
```diff
@@ -125,6 +126,46 @@ public Source resolve(final String href,
...
+            "<?xml-stylesheet href='/b.xsl' type='text/xsl'?>",
+            "<subject>World</subject>"
+        );
+        final String xsl = Joiner.on(' ').join(
+            "<stylesheet xmlns='http://www.w3.org/1999/XSL/Transform'  ",
+            " xmlns:x='http://www.w3.org/1999/xhtml' version='2.0'  >",
+            "<output method='text'/><template match='/'> ",
+            "Hello, <value-of select='/subject'/>!</template></stylesheet>"
+        );
+        final StateAwareInputStream inputStream =
```
Comment (yegor256) [inline src/test/java/org/takes/rs/RsXSLTTest.java]: @Happy-Neko just `stream` is not enough? see http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html

**Comment by yegor256** [comment]
@Happy-Neko one small comment from me above

**Comment by yegor256** [comment]
@Happy-Neko I like this design, but I wonder wether we're the first who needs this... maybe there are some existing libraries out there that provide such `StateAwareInputStream` class (or a similar one).. just curious.

**Comment by Happy-Neko** [comment]
@yegor256 `inputStream` variable renamed to `stream`.
I made quick search through guava and commons-io before but didnt find anything like `StateAwareInputStream`.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@ypshenychka please, review this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by ypshenychka** [comment]
@carlosmiranda 
According to our [QA Rules](http://www.teamed.io/qa.html):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Two issues were found during code review, one of which is cosmetic.
Please confirm that you'll try to find at least three major problems while future reviews along with cosmetic concerns.

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed, thanks

**Comment by ypshenychka** [comment]
@carlosmiranda thank you

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
@carlosmiranda 10 mins was added to the account of @ypshenychka (for QA review), in transaction 72821286; done, I added **20 mins** in payment `72821296`, 47 hours and 4 mins spent to complete this; review comments (c=5) added as a bonus; +20 to your rating, your total score is [+2144](http://www.netbout.com/b/35023?open=rating)

**Pull request merged**